src/contacts/tests/test_contacts.py

```diff
@@ -0,0 +1,333 @@
+                    "PUBLIC_KEY": "1",
+                },
+            ):
+                # mock db module as MagicMock, context manager handles cleanup
+                with patch("contacts.contacts.ContactsDb") as mock_db:
+                    self.mocked_db = mock_db
+                    # get create flask app
+                    self.flask_app = create_app()
+                    # set testing config
+                    self.flask_app.config["TESTING"] = True
+                    # create test client
+                    self.test_app = self.flask_app.test_client()
+                    # set public key
+                    self.flask_app.config["PUBLIC_KEY"] = EXAMPLE_PUBLIC_KEY
+                    # mock return value of get_contacts to return empty
+                    self.mocked_db.return_value.get_contacts.return_value = []
+
+    def test_version_endpoint_returns_200_status_code_correct_version(self):
+        """test if correct version is returned"""
+        # generate a version
+        version = str(random.randint(1, 9))
+        # set version in Flask config
+        self.flask_app.config["VERSION"] = version
+        # send get request to test client
+        response = self.test_app.get("/version")
+        # assert 200 response code
+        self.assertEqual(response.status_code, 200)
+        # assert both versions are equal
+        self.assertEqual(response.data, version.encode())
+
+    def test_ready_endpoint_200_status_code_ok_string(self):
+        """test if correct response is returned from readiness probe"""
+        response = self.test_app.get("/ready")
+        # assert 200 response code
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data, b"ok")
+
+    def test_create_contact_201_status_code_correct_db_contact_object(self):
+        """test adding a new contact to a users contact list"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
+        # send request to test client
+        response = self.test_app.post(
+            "/contacts/{}".format(EXAMPLE_USER),
+            headers=EXAMPLE_HEADERS,
+            data=json.dumps(example_contact),
+        )
+        # assert 201 response code
+        self.assertEqual(response.status_code, 201)
+        # assert contact object added to database had the required fields
+        # get the arg that contact_db.add_contact was called with
+        contact_object = self.mocked_db.return_value.add_contact.call_args[0][0]
+        # add username to example contact object
+        example_contact["username"] = EXAMPLE_USER
+        # assert all keys are equal
+        self.assertEqual(contact_object, example_contact)
+
+    def test_create_contact_409_status_code_add_same_user_to_contacts(self):
+        """test adding a contact with same account_num and routing_num as the user"""
+        # create example contact request
+        example_contact = EXAMPLE_CONTACT.copy()
```

Comment:
changing var name to `invalid_contact` could be more readable

Reply:
Done